test_util/src/lib.rs

```diff
@@ -338,6 +360,55 @@ async fn run_wss_server(addr: &SocketAddr) {
   }
 }
 
+/// This server responds with 'PASS' if client authentication was successful. Try it by running
+/// test_server and
+///   curl --key cli/tests/tls/localhost.key \
+///        --cert cli/tests/tls/localhost.crt \
+///        --cacert cli/tests/tls/RootCA.crt https://localhost:4552/
+async fn run_tls_client_auth_server() {
+  let cert_file = "cli/tests/tls/localhost.crt";
+  let key_file = "cli/tests/tls/localhost.key";
+  let ca_cert_file = "cli/tests/tls/RootCA.pem";
+  let tls_config = get_tls_config(cert_file, key_file, ca_cert_file)
+    .await
+    .unwrap();
+  let tls_acceptor = TlsAcceptor::from(tls_config);
+
+  // This bit of nastiness is to support the case when localhost resolves to
+  // ::1 on github actions and 127.0.0.1 locally.
```

Comment:
This is necessary because cli/tests/tls/localhost.crt is made out to "localhost", right? I'd make that explicit in the comment, it's not self-evident.

I'd bind to all addresses instead of just the first one, otherwise you might run into the issue where the server picks 127.0.0.1 but the client ::1, or vice versa. Example code:
```rust
  //use futures::FutureExt;
  //use futures::StreamExt;

  let accept = |listener: tokio::net::TcpListener| {
    async {
      let result = listener.accept().await;
      Some((result, listener))
    }
    .boxed_local()
  };

  let host_and_port = &format!("localhost:{}", TLS_CLIENT_AUTH_PORT);

  let listeners = tokio::net::lookup_host(host_and_port)
    .await
    .expect(host_and_port)
    .inspect(|address| println!("{} -> {}", host_and_port, address))
    .map(tokio::net::TcpListener::bind)
    .collect::<futures::stream::FuturesUnordered<_>>()
    .collect::<Vec<_>>()
    .await
    .into_iter()
    .filter_map(Result::ok) // TODO(bnoordhuis) Log errors.
    .map(|listener| futures::stream::unfold(listener, accept))
    .collect::<Vec<_>>();

  let mut listeners = futures::stream::select_all(listeners);

  while let Some(Ok((stream, _addr))) = listeners.next().await {
    // ...
  }
```

Reply:
Added. Thanks!